Remotion/Data/DomainObjects.UnitTests/Mapping/TypeDefinitionChecker.cs

```diff
@@ -61,29 +68,20 @@ public void Check (TypeDefinition expectedDefinition, TypeDefinition actualDefin
 
       foreach (var expectedDefinition in expectedDefinitions)
       {
-        var actualDefinition = actualDefinitions[expectedDefinition.Type];
-        if (expectedDefinition is ClassDefinition expectedClassDefinition)
-        {
-          if (actualDefinition is not ClassDefinition actualClassDefinition)
-          {
-            Assert.Fail("The two type definitions must have a matching type.");
-            return;
-          }
-
-          CheckClassDefinition(expectedClassDefinition, actualClassDefinition);
-          CheckDerivedClasses(expectedClassDefinition, actualClassDefinition);
-        }
-        else
+        if (!actualDefinitions.TryGetValue(expectedDefinition.Type, out var actualDefinition))
         {
-          Assert.Fail("Unsupported type definition cannot be checked.");
+          Assert.Fail("The two type definitions must have a matching type.");
```

Comment:
we should provide the names for diagnostic reasons. not fun, having this without infos.